src/field/FieldEquation.hh

```diff
@@ -0,0 +1,52 @@
+//---------------------------------*-CUDA-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file FieldEquation.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "MagField.hh"
+#include "base/Types.hh"
+#include "physics/base/Units.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * The FieldEquation evaluates the righ hand side of the Lorentz equation
+ * for a given magnetic field value.
+ */
+class FieldEquation
+{
+  public:
+    //@{
+    //! Type aliases
+    using OdeState = Array<real_type, 6>;
+    //@}
+
+  public:
+    //! Construct with a magnetic field
+    CELER_FUNCTION FieldEquation(const MagField& field);
+
+    //! Set charge if q != -eplus
+    CELER_FUNCTION void set_charge(units::ElementaryCharge q);
+
+    //! Evaluate the right hand side of the field equation
+    CELER_FUNCTION auto operator()(const OdeState& y) const -> OdeState;
+
+  private:
+    //! Scale factor for the coefficient of the equation (temporary)
+    static CELER_CONSTEXPR_FUNCTION real_type scale() { return 1e-14; }
```

Comment:
Especially since we're breaking up the original mega-magnetic-field PR into multiple smaller requests, please replace this mystery constant with the native unit system (or suitable usage of `Quantity` that would let the class internally operate in whatever unit system it needs).

Reply:
This is a just a conversion factor for units (as we use SI unit for the magnetic field and charge).  I have not figured out the better implementation, but renamed it to unit_conversion.  Any suggestion?

Reply:
I'm not sure I understand what the scaling is needed for -- I think maybe you should be converting the charge to the native unit system:
```c++
coeffi_ = unit_cast(charge_) * constants::c_light;
```
which converts the charge from elemental to Colombs so it can work with the native units for the magnetic field (specified with the `telsa` constant).

Reply:
Thanks.  Changed the coefficient with celeritas units:
coeffi = unit_cast(charge_) /unit_cast{MevMomentum{1.0};

Reply:
Aha, that's where the other factor is coming from! Since the non-native units of the momentum field are embedded into the values of the `FieldState` (or whatever we're calling it), can you add to the that class:
```c++
using Momentum = MevMomentum;
```
and then use
```c++
unit_cast(charge_) / FieldState::Momentum::value();
```
to safely propagate the nonstandard unit types?

Reply:
Yes, a good suggestion - it may be inside FieldTrackView (has not introduced yet in the MR, but will be included in the next set) where OdeState is initialized (with native celeritas units, i.e, length in cm, momentum in MevMomentum, and the charge in ElementaryCharge).  For this MR, I may put this inside FieldInterface which I additionally introduce for the OdeState definition and will move it to  FieldTrackView later.

Reply:
Understood. As long as we can track the different units throughout the codebase (so that someone can tell what units to use with a class just by looking at its interface) I'll be satisfied 😄